dhis-2/dhis-api/src/main/java/org/hisp/dhis/category/CategoryService.java

```diff
@@ -520,4 +520,9 @@ CategoryOptionCombo getCategoryOptionCombo( CategoryCombo categoryCombo,
     CategoryOption getDefaultCategoryOption();
 
     Category getDefaultCategory();
+
+    Category getDefaultCaEagerFetch();

```

Comment:
Let's move these methods to where they belong in terms of order, e.g. getDefaultCcEagerFetch under CategoryCombo section, getDefaultCoCEagerFetch under CategoryOptionCombo section, etc.

Reply:
I'm not a big fan on `getters` (or mutators) that have a non standard semantic: as a developer, when I'm supposed to use this method? 
If you think that it's better to fetch the Category eagerly, I'd rather change the Hibernate mapping. I understand you have to do this, because in certain scenarios you were getting the dreaded `NoSessionAvailable` from Hibernate: I think we can try to understand how to mitigate this rather than introducing new companion methods.

Reply:
Changing Hibarnate mapping will always eagerly fetch collections and this is also something which we don't want.
Secondly specifically for this issue if we find a way not to cache hibernate objects then we don't need these methods.

Reply:
The problem is that _most_ of the time we do *not* want to eager load associations, as it will be terrible for performance.

Perhaps the real problem to fix is why and how we cache these default objects, which could have been done differently in IdentifiableObjectManager.